src/bars/index.tsx

```diff
@@ -69,35 +70,35 @@ export class AudioBars extends Component<AudioBarsProps, {}> {
   }
 
   private drawBars = (): void => {
-    const {canvasContext, width: canvasWidth, height: canvasHeight, dataArray} = this;
-
+    const {canvasContext, barWidth, numBarsToDraw, dataArray, width: canvasWidth, height: canvasHeight} = this;
+    const {analyser} = this.props;
     // clear the canvas
-    this.canvasContext.clearRect(0, 0, canvasWidth, canvasHeight);
+    // TODO [Perf] we can store the higher bar value XY and only clear the canvas from those points
+    canvasContext.clearRect(0, 0, canvasWidth, canvasHeight);
 
-    const {analyserNode} = this.props.analyser;
+    const {analyserNode} = analyser;
     const maxByteValue = 256;
     analyserNode.getByteFrequencyData(dataArray);
 
-    const numBarsToDraw = Math.min(dataArray.length, 64);
     const bufferLength = dataArray.length;
 
     // chunk values if too many to display
     const numValuesPerChunk = Math.ceil(bufferLength / numBarsToDraw);
     const chunkedData = chunk(dataArray, numValuesPerChunk);
     const barValues = chunkedData.map((arr: Array<number>) => sum(arr) / arr.length);
 
-    const barWidth = canvasWidth / numBarsToDraw;
-
     // draw the bars
     for (let i = 0; i < barValues.length; i++) {
       const x = i * barWidth + i;
 
+      if (x >= canvasWidth) {break;} // Don't paint bars outside the canvas
```

Comment:
It was happening before for sure, I was logging the ```x``` and it was bigger than the ```canvasWidth```, not sure if still happens with my changes